src/org/benf/cfr/reader/bytecode/analysis/opgraph/op4rewriters/RecordRewriter.java

```diff
@@ -333,18 +337,22 @@ private static boolean removeImplicitAssignments(Method canonicalCons, List<Clas
             // this is very messy - refactor using wildcardmatch.
             if (statement.isEffectivelyNOP()) continue;
             if (!(statement instanceof StructuredAssignment)) break;
-            LValue lhs = ((StructuredAssignment) statement).getLvalue();
+
+            StructuredAssignment assignment = (StructuredAssignment) statement;
+            LValue lhs = assignment.getLvalue();
             ClassFileField field = getCFF(lhs, thisType);
             if (field == null) break;
             int idx = instances.indexOf(field);
             if (idx == -1) break;
             instances.set(idx, null);
 
-            Expression rhs = ((StructuredAssignment) statement).getRvalue();
+            Expression rhs = assignment.getRvalue();
             if (!(rhs instanceof LValueExpression)) break;
             LValue rlv = ((LValueExpression) rhs).getLValue();
             LocalVariable expected = args.get(idx);
             if (rlv != expected) break;
+            // Implicit parameter must use the same name as the field
+            expected.getName().forceName(field.getFieldName());
```

Comment:
Might have to be moved down outside the loop to be sure that this is actually a compact canonical constructor. But probably does not hurt to have it here either, since the condition in the line before already made sure that parameter and field match.